server/src/test/java/io/crate/expression/scalar/SubscriptFunctionTest.java

```diff
@@ -69,9 +69,7 @@ public void testIndexOutOfRange() throws Exception {
 
     @Test
     public void testIndexExpressionIsNotInteger() throws Exception {
-        expectedException.expectMessage(
-            "`index` in subscript expression (`base[index]`) " +
-            "must be a numeric type if the base expression is text_array");
+        expectedException.expectMessage("No cast function found for return type undefined");
```

Comment:
I think this error message is not really understandable for users. Is it possible to improve here?

Reply:
probably cannot do it better than this https://github.com/crate/crate/pull/9928/files#diff-fb3a57a840597e5dc05aa82f387d13e6R406, otherwise, we will get the casting exception message.

Reply:
👍 Think this is much better now.